File: src/fe/src/models/WorklistFromBioHubItemMaterial.ts
import { Gender } from "@/models/enums/Gender";
import { ShipmentMaterialCondition } from "@/models/enums/ShipmentMaterialCondition";
import { MaterialStatus } from "./enums/MaterialStatus";

export interface WorklistFromBioHubItemMaterial {
  Id: string;
  WorklistFromBioHubItemId: string;
  MaterialId: string;
  Quantity: number;
  AvailableQuantity: number;
  Amount: number;
  MaterialNumber: string;
  MaterialProductId: string;
  TransportCategoryId: string;
  MaterialName: string;
  CollectionDate: Date;
  Location: string;
  IsolationHostTypeId: string;
  Gender: Gender;
  Age: number;
  Condition: ShipmentMaterialCondition;
  Note: string;
  Status: MaterialStatus;
}

File: src/be/src/WHO.BioHub.Models/Models/WorklistFromBioHubItemMaterial.cs
using WHO.BioHub.Models.Models.Abstractions;
using WHO.BioHub.Shared.Enums;

namespace WHO.BioHub.Models.Models;

public class WorklistFromBioHubItemMaterial : EntityBase
{
    public Guid? WorklistFromBioHubItemId { get; set; }
    public WorklistFromBioHubItem WorklistFromBioHubItem { get; set; }
    public Guid? MaterialId { get; set; }
    public Material Material { get; set; }
    public int? Quantity { get; set; }
    public double? Amount { get; set; }
    public ShipmentMaterialCondition? Condition { get; set; }
    public string Note { get; set; }
}

